src/renderer/views/Detail/Detail.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { Section, Label, Input } from '../../components/Setting/Setting';
+import Button from '../../elements/Button/Button';
+import { RootState } from '../../store/reducers';
+import appStyles from '../../App.module.css';
+import * as LibraryActions from '../../store/actions/LibraryActions';
+import * as coverUtils from '../../../shared/lib/utils-cover';
+import Placeholder from '../../../images/assets/placeholder.png';
+import { parseDuration } from '../../lib/utils';
+
+import styles from './Detail.module.css';
+
+const Detail: React.FC = () => {
+  const { trackId } = useParams<{ trackId: string }>();
+  const tracks = useSelector((state: RootState) => state.library.tracks.library);
+  const [track, setTrack] = useState(undefined);
+  const [coverSrc, setCoverSrc] = useState(undefined);
+  const [data, setData] = useState({
+    title: '',
+    artist: '',
+    album: '',
+    genre: '',
+    duration: '',
+  });
+
+  const getCover = async (path: string) => {
+    const data = await coverUtils.fetchCover(path);
+    if (data !== null) {
+      setCoverSrc(data);
+    }
+  };
+
+  const updateSong = (data) => {
+    const song = { ...track };
+    song.title = data.title;
+    song.artist[0] = data.artist;
+    song.album = data.album;
+    song.genre[0] = data.genre;
+    return song;
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const song = updateSong(data);
+    await LibraryActions.updateTrack(song);
+    // dispatch(LibraryActions.refresh());
+    history.back();
+  };
+
+  const handleChange = (e) => {
+    e.preventDefault();
+    setData({
+      ...data,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleCancel = (e) => {
+    e.preventDefault();
+    history.back();
+  };
+
+  useEffect(() => {
+    const song = tracks.find((tr) => tr._id === trackId);
```

Comment:
Maybe you can use here the `db.tracks` object instead of the data coming from the store? wdyt?